crates/sui/src/client_commands.rs

```diff
@@ -383,6 +408,38 @@ impl SuiClientCommands {
                 SuiClientCommandResult::TransferSui(cert, effects)
             }
 
+            SuiClientCommands::Pay {
+                input_coins,
+                recipients,
+                amounts,
+                gas,
+                gas_budget,
+            } => {
+                ensure!(
```

Comment:
Should we also protect against empty recipients, mismatched recipient/amount lengths, etc?

Reply:
here I am trying to avoid the potential panic caused by `input_coins[0]`, we now have these checks including the emtpy input coins in pay() in execution_engine.rs, do we want to move all those checks here to catch them earlier?